pypeit/spectrographs/lbt_luci.py

```diff
@@ -360,30 +394,115 @@ def default_pypeit_par(cls):
+
 
         return par
 
-# TODO: OUT OF DATE
-#    def check_headers(self, headers):
-#        """
-#        Check headers match expectations for an LBT LUCI1 exposure.
-#
-#        See also
-#        :func:`pypeit.spectrographs.spectrograph.Spectrograph.check_headers`.
-#
-#        Args:
-#            headers (list):
-#                A list of headers read from a fits file
-#        """
-#        expected_values = { '0.INSTRUME': 'LUCI1',
-#                            '0.NAXIS': 2 }
-#        super(LBTLUCI1Spectrograph, self).check_headers(headers,
-#                                                              expected_values=expected_values)
+    def get_rawimage(self, raw_file, det):
+        """
+        Read raw images and generate a few other bits and pieces that are key
+        for image processing.
+
+        .. warning::
+
+            - When reading multiple detectors for a mosaic, this function
+              expects all detector arrays to have exactly the same shape.
+
+        Parameters
+        ----------
+        raw_file : :obj:`str`
+            File to read
+        det : :obj:`int`, :obj:`tuple`
+            1-indexed detector(s) to read.  An image mosaic is selected using a
+            :obj:`tuple` with the detectors in the mosaic, which must be one of
+            the allowed mosaics returned by :func:`allowed_mosaics`.
+
+        Returns
+        -------
+        detector : :class:`~pypeit.images.detector_container.DetectorContainer`, :class:`~pypeit.images.mosaic.Mosaic`
+            Detector metadata parameters for one or more detectors.
+        raw : `numpy.ndarray`_
+            Raw image for this detector.  Shape is 2D if a single detector image
+            is read and 3D if multiple detectors are read.  E.g., the image from
+            the first detector in the tuple is accessed using ``raw_img[0]``.
+        hdu : `astropy.io.fits.HDUList`_
+            Opened fits file
+        texp : :obj:`float`
+            Exposure time *in seconds*.
+        rawdatasec : `numpy.ndarray`_
+            Data (Science) section of the detector as provided by setting the
+            (1-indexed) number of the amplifier used to read each detector
+            pixel. Pixels unassociated with any amplifier are set to 0.  Shape
+            is identical to ``raw_img``.
+        oscansec : `numpy.ndarray`_
+            Overscan section of the detector as provided by setting the
+            (1-indexed) number of the amplifier used to read each detector
+            pixel. Pixels unassociated with any amplifier are set to 0.  Shape
+            is identical to ``raw_img``.
+        """
+
+        detector, raw, hdu, texp, datasec, oscansec = super().get_rawimage(raw_file, det)
+
+        # Non-linearity correction
+        # See: https://scienceops.lbto.org/luci/instrument-characteristics/detector/
+        # I assume that the correction applies to each DIT, not the full exposure.
+        ndit = hdu[0].header['NDIT']
```

Comment:
I'm not opposed to this, but I assume we don't have the meta info at this point to grab NDIT... Also, Is this the correct place to perform the non-linearity correction? I think the non-linearity should come after the bias/dark subtraction. See `pypeit.images.rawimage.py`